tests/integration/utilities/test_prometheus_collectors.py

```diff
@@ -29,6 +32,41 @@ def mock_operator_confirmation(random_address, mock_taco_application_agent):
     mock_taco_application_agent.get_staking_provider_info.return_value = info
 
 
+def test_start_prometheus_exporter_called(ursulas, mock_prometheus):
+    port = 9101
+
+    # Reset start_prometheus_exporter mock just in case it was previously called
+    mock_prometheus.reset_mock()
```

Comment:
I'm not a Python scientist, but would it be possible to move this call into the `mock_prometheus` fixture or is there some reason not to do this? Just curious

Reply:
Is it necessary for the fixture to be "session"-scoped?

For some of these tests perhaps you can instead use a `spy`?

Reply:
> Is it necessary for the fixture to be "session"-scoped?
> 
> For some of these tests perhaps you can instead use a `spy`?

This was added because there are some tests that run several Ursula's instances. So, since Prometheus exporter now runs with each Ursula instance, those tests also try to run several Prometheus exporter instances. This causes test errors because each  Prometheus instance tries to lock the same resources (network port). 

Also, Prometheus exporter is not necessary for most tests, and running it can cause errors, a waste of resources, and tests running for more time. So I decided to just mock it.

Does it make sense?

More info in this commit: 
https://github.com/nucypher/nucypher/pull/3294/commits/9c1547ef5e507b5ba32c151ad83f4252a8982061